src/components/source_editor/LatticeConfigurationDialog.js

```diff
@@ -136,6 +146,7 @@ class LatticeConfigurationDialog extends React.Component {
                                 <label>Lattice type</label>
                                 <div className="fg-line">
                                     <select label="Lattice type" name="type" className="form-control fc-alt"
+                                        disabled={this.isDisabled(this.state.isNonPeriodic)}
```

Comment:
I suggest not touching the existing functionality at first and introducing minimal changes if at all to Lattice/Basis